apps/dav/lib/CalDAV/Status/StatusService.php

```diff
@@ -123,19 +124,21 @@ public function processCalendarStatus(string $userId): void {
 			return;
 		}
 
-		// One event that fulfills all status conditions is enough
-		// 1. Not an OOO event
-		// 2. Current user status was not set after the start of this event
-		// 3. Event is not set to be transparent
-		$count = count($applicableEvents);
-		$this->logger->debug("Found $count applicable event(s), changing user status", ['user' => $userId]);
-		$this->userStatusService->setUserStatus(
-			$userId,
-			IUserStatus::AWAY,
-			IUserStatus::MESSAGE_CALENDAR_BUSY,
-			true
-		);
-
+		// Only update the status if it's neccesary otherwise we mess up the timestamp
+		if($currentStatus !== null && $currentStatus->getMessageId() !== IUserStatus::MESSAGE_CALENDAR_BUSY) {
```

Comment:
what would be messed up?

Reply:
I've had that in my local env. The timestamp is updated, thus the DTSTART is lower than the timestamp - which leads to a reset because no event is found (which is wrong).

Reply:
One of the unit tests fails, so I wonder if the test is actually right and the code here wrong?
Should it be:
```suggestion
		// Only update the status if it's neccesary otherwise we mess up the timestamp
		if ($currentStatus === null || $currentStatus->getMessageId() !== IUserStatus::MESSAGE_CALENDAR_BUSY) {
```

Otherwise the event status will only be set, when you had any status before?